tiledb/sm/array_schema/array_schema.cc

```diff
@@ -538,6 +554,62 @@ Status ArraySchema::drop_attribute(const std::string& attr_name) {
   return Status::Ok();
 }
 
+Status ArraySchema::rename_attribute(
+    const std::string& old_name,
+    const std::string& new_name,
+    bool is_new_uri = true) {
+  std::lock_guard<std::mutex> lock(mtx_);
+  if (old_name.empty() || new_name.empty()) {
+    // Do nothing if name is empty
+    return Status::Ok();
+  }
+
+  auto it = attribute_map_.find(old_name);
+  if (it != attribute_map_.end()) {
+    it->second->set_name(new_name);
+    attribute_map_[new_name] = std::move(it->second);
+    attribute_map_.erase(it);
+  }
+
+  // Save the old name
+  attribute_used_name_map_[old_name] = new_name;
+
+  for (auto it = attribute_used_name_map_.begin();
+       it != attribute_used_name_map_.end();
+       ++it) {
+    if (it->second == old_name) {
+      attribute_used_name_map_[it->first] = new_name;
+    }
+  }
+
+  if (is_new_uri) {
+    RETURN_NOT_OK(generate_uri());
+  }
+
+  return Status::Ok();
+}
+
+std::vector<std::string> ArraySchema::attribute_used_names() const {
+  std::vector<std::string> result;
+  std::lock_guard<std::mutex> lock(mtx_);
+  result.reserve(attribute_used_name_map_.size());
+  for (auto it = attribute_used_name_map_.begin();
+       it != attribute_used_name_map_.end();
+       ++it) {
+    result.push_back(it->first);
+  }
+  return result;
+}
+
+std::string ArraySchema::attribute_current_name(
+    const std::string& used_name) const {
+  std::lock_guard<std::mutex> lock(mtx_);
+  return (attribute_used_name_map_.find(used_name) ==
+          attribute_used_name_map_.end()) ?
+             "" :
```

Comment:
Is this ok to be "unsafe"? Do we want to change the function to return `Status` in case that `used_name` does not exist?

Reply:
Changed to return an error status if old_name does not exist